feature/bgp/otg_tests/bgp_override_as_path_split_horizon_test/bgp_override_as_path_split_horizon_test.go

```diff
@@ -0,0 +1,581 @@
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	otgtelemetry "github.com/openconfig/ondatra/gnmi/otg"
+	otg "github.com/openconfig/ondatra/otg"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+const (
+	advertisedRoutesv4CIDR   = "203.0.113.1/32"
+	advertisedRoutesv4Net    = "203.0.113.1"
+	advertisedRoutesv4Prefix = 32
+	peerGrpName1             = "BGP-PEER-GROUP1"
+	peerGrpName2             = "BGP-PEER-GROUP2"
+	dutGlobalAS              = 64512
+	dutLocalAS1              = 65501
+	dutLocalAS2              = 64513
+	ateAS1                   = 65502
+	ateAS2                   = 65503
+	plenIPv4                 = 30
+	plenIPv6                 = 126
+	policyName               = "ALLOW"
+)
+
+var (
+	dutPort1 = attrs.Attributes{
+		Desc:    "DUT to ATE Port1",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::192:0:2:1",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	atePort1 = attrs.Attributes{
+		Name:    "atePort1",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		MAC:     "02:00:01:01:01:01",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	dutPort2 = attrs.Attributes{
+		Desc:    "DUT to ATE Port2",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::192:0:2:5",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+	atePort2 = attrs.Attributes{
+		Name:    "atePort2",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::192:0:2:6",
+		MAC:     "02:00:02:01:01:01",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+
+	nbr1 = &bgpNeighbor{localAS: dutLocalAS1, peerAS: ateAS1, neighborip: atePort1.IPv4, isV4: true, peerGrp: peerGrpName1}
+	nbr2 = &bgpNeighbor{localAS: dutLocalAS2, peerAS: ateAS2, neighborip: atePort2.IPv4, isV4: true, peerGrp: peerGrpName2}
+
+	otgPort1V4Peer = "atePort1.BGP4.peer"
+	otgPort2V4Peer = "atePort2.BGP4.peer"
+)
+
+// configureDUT configures all the interfaces on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	dc := gnmi.OC()
+	i1 := dutPort1.NewOCInterface(dut.Port(t, "port1").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	i2 := dutPort2.NewOCInterface(dut.Port(t, "port2").Name(), dut)
+	gnmi.Replace(t, dut, dc.Interface(i2.GetName()).Config(), i2)
+}
+
+// verifyPortsUp asserts that each port on the device is operating.
+func verifyPortsUp(t *testing.T, dev *ondatra.Device) {
+	t.Helper()
+	for _, p := range dev.Ports() {
+		status := gnmi.Get(t, dev, gnmi.OC().Interface(p.Name()).OperStatus().State())
+		if want := oc.Interface_OperStatus_UP; status != want {
+			t.Errorf("%s Status: got %v, want %v", p, status, want)
+		}
+	}
+}
+
+// bgpCreateNbr creates a BGP object with neighbors pointing to atePort1 and atePort2
+func bgpCreateNbr(t *testing.T, dut *ondatra.DUTDevice) *oc.NetworkInstance_Protocol {
```

Comment:
This should use cfgplugins.NewBGPSession. 

For an example, see:
https://github.com/openconfig/featureprofiles/blob/520108e9704e82444726ff35d0687f09163f8994/feature/bgp/policybase/otg_tests/aspath_test/aspath_test.go#L221

Reply:
@dplore  will update all review comments . thanks

Reply:
@dplore  I did check cfgplugins and its very generic one. We need to have specific AS numbers for this test. Please let me know if is ok to have local config and validation functions for this test.

Reply:
Ok.